example-client/client.py

```diff
@@ -67,40 +69,42 @@ def run_structure(input: str) -> Optional[str]:
     # Runs are asynchronous, so we need to poll the status until it's no longer running.
     structure_run_id = structure_run["structure_run_id"]
     status = structure_run["status"]
-    printed_event_ids = set()  # Keep track of which events we've printed.
-    while status not in ("SUCCEEDED", "FAILED"):
-        structure_run = get_structure_run(
-            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
+    offset = 0
+    output = None
+    while not is_status_complete(status) or output is None:
+        event_response = get_structure_run_events(
+            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id, offset=offset
         )
-        status = structure_run["status"]
-
-        # You can comment out this block if you don't want to stream events as they occur.
-        event_list = get_structure_run_events(
-            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
-        )
-        events = event_list["events"]
-        printed_event_ids = print_streaming_events(events, printed_event_ids)
-
+        events = event_response["events"]
+        for event in events:
+            if event["origin"] == "SYSTEM":
```

Comment:
any reason the user has to care about the origin?

Reply:
the origin is how system events are determined from events that originate in the container. we want to make sure there is no name conflicts as the user events could be anything.

Reply:
Is it clearly specified anywhere what events are SYSTEM and what are USER?

Reply:
Yeah, I'm not sure the user needs to care about the provenance of the event, they care about the event. And agreed that "USER" is confusing.

Reply:
`USER` in the context of griptape cloud means anything that originates from structure code. `SYSTEM` is any event not from structure code. i removed the origin check from the example.

Reply:
The distinction between origin types is going to be in our documentation for users to reference: https://github.com/griptape-ai/griptape/pull/1187